From e70c839fb7f129eb56e12ba66f380e3b2d4a5777 Mon Sep 17 00:00:00 2001
Message-Id: <e70c839fb7f129eb56e12ba66f380e3b2d4a5777.1424394676.git.feitong.yi@intel.com>
In-Reply-To: <00f4430a265007e8e2c016cc7926bf530835ea30.1424394676.git.feitong.yi@intel.com>
References: <00f4430a265007e8e2c016cc7926bf530835ea30.1424394676.git.feitong.yi@intel.com>
From: Rafael Barbalho <rafael.barbalho@intel.com>
Date: Wed, 11 Feb 2015 13:57:51 +0000
Subject: [PATCH 55/61] FOR_UPSTREAM [VPG]: drm/i915: Correct the size of PDE
 allocation

Use the correct parameter when calling alloc pages.

For: GMINL-5925
Change-Id: I3561ef6b4a539b1054eebbb4ef1368647bbe84ba
Signed-off-by: Rafael Barbalho <rafael.barbalho@intel.com>
---
 drivers/gpu/drm/i915/i915_gem_gtt.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_gem_gtt.c b/drivers/gpu/drm/i915/i915_gem_gtt.c
index d18fcf8..3f33d53 100644
--- a/drivers/gpu/drm/i915/i915_gem_gtt.c
+++ b/drivers/gpu/drm/i915/i915_gem_gtt.c
@@ -496,7 +496,8 @@ static int gen8_ppgtt_allocate_page_directories(struct i915_hw_ppgtt *ppgtt,
 	* pre-fetches entries; that pre-fetch can attempt access for entries
 	* even if no resources are located in that range.
 	*/
-	ppgtt->pd_pages = alloc_pages(GFP_KERNEL, GEN8_LEGACY_PDPS);
+	ppgtt->pd_pages = alloc_pages(GFP_KERNEL,
+				get_order(GEN8_LEGACY_PDPS << PAGE_SHIFT));
 	if (!ppgtt->pd_pages) {
 		__free_page(ppgtt->scratch_page);
 
-- 
1.7.9.5

